docs/source/data_entry.rst

```diff
@@ -137,39 +137,75 @@ Setting ``dry_run=False`` is equivalent to not specifying ``dry_run`` at all and
+    process_spec = ProcessSpec(name='spec name')
+    process_run = ProcessRun(name='run name', spec=process_spec, conditions=[condition1, condition2], parameters=[parameter1, parameter2])
+    dataset.process_runs.validate_templates(process_run)
 
-Example usage with an object template:
+has return value:
 
 .. code-block:: python
 
-    template = ProcessTemplate("pt")
-    spec = ProcessSpec("foo", template=template)
-    run = ProcessRun("bar", spec=spec)
-    dataset.process_runs.validate_templates(run, object_template=template)
+    [{'failure_message': 'Multiple Condition with named condition_name', 'property': None, 'failure_id': 'attribute.duplicate', 'input': None, 'type': NotImplemented},
+     {'failure_message': 'Multiple Parameter with named parameter_name', 'property': None, 'failure_id': 'attribute.duplicate', 'input': None, 'type': NotImplemented}]
+
+Example with validation errors with an object template:
+
+.. code-block:: python
+
+    condition_template = ConditionTemplate("condition template", bounds=IntegerBounds(1, 5))
+    condition_template = dataset.condition_templates.register(condition_template)
+
+    condition = Condition("condition", value=UniformInteger(1, 3), template=condition_template)
+    process_template = ProcessTemplate("pt", conditions=[[LinkByUID("id", condition_template.uids["id"]), IntegerBounds(2, 4)]])
+    process_spec = ProcessSpec("ps", template=process_template)
+    process_run = ProcessRun("pr", conditions=[condition], spec=process_spec)
+    dataset.process_runs.validate_templates(process_run, object_template=process_template)
+
+has return value:
 
-Example usage for an ingredient:
+.. code-block:: python
+
+    [{'failure_message': 'UniformInteger(1,3) extends below 2 {2}', 'property': None, 'failure_id': 'attribute.bounds.value', 'input': None, 'type': NotImplemented}]
+
+For ingredients, the associated object template is a process template that is provided as a separate parameter. Any
+value provided to the ``object_template`` parameter for an ingredient will be ignored.
+
+Example with validation errors for an ingredient:
 
 .. code-block:: python
 
-    process_template = ProcessTemplate("pt")
+    process_template = ProcessTemplate("pt", allowed_names=["foo"], allowed_labels=["bar"])
     process_spec = ProcessSpec("ps", template=process_template)
-    material_spec = MaterialSpec("ms", process=process_spec)
-    ingredient_spec = IngredientSpec("is", process=process_spec, material=material_spec)
-    dataset.ingredient_specs.validate_templates(run, ingredient_process_template=process_template)
+
+    mat_process_spec = ProcessSpec("mps")
+    material_spec = MaterialSpec("ms", process=mat_process_spec)
+
+    ingredient_spec = IngredientSpec("is", process=process_spec, material=material_spec, labels=["ingredient"])
+    dataset.ingredient_specs.validate_templates(ingredient_spec, ingredient_process_template=process_template)
```

Comment:
If the collection is ingredient specs/runs, is the `object_template` field ignored?

Reply:
Sort of? It's passed through to the core validation code, which then validates the object template against all attributes that have attribute templates that are also on the object template. But ingredients don't have attributes, so there's nothing to validate the object template against